src/Nethermind/Nethermind.Synchronization/BeamSync/BeamBlockchainProcessor.cs

```diff
@@ -366,9 +375,9 @@ private Task PrefetchNew(IStateReader stateReader, Block block, Keccak stateRoot
             {
                 BeamSyncContext.MinimumDifficulty.Value = block.TotalDifficulty ?? 0;
                 BeamSyncContext.Cancelled.Value = cancellationToken.Token;
-                for (int i = 0; i < block.Transactions.Length; i++)
+                for (int i = 0; i < (block.Transactions?.Length ?? 0); i++)
                 {
-                    Transaction tx = block.Transactions[i];
+                    Transaction tx = block.Transactions![i];
```

Comment:
Another interesting one.  My personal strategy is to never allow collections to be null and instead prefer an empty collection.  However, that felt like a big change which I wasn't sure how you all would feel about so I went with just fixing the loop to loop 0 times in such cases instead.

Reply:
I would like to make block.Transactions property non-nullable

Reply:
yes, big change, part of the nullability work which I kept for holidays (the kind of work that is so satisfying t work on that I can pick it instead of holidays)